tests/test_evalscript.py

```diff
@@ -1,105 +1,100 @@
+    expected_bands_str = "bands: [" + ", ".join(f'"{b}"' for b in all_bands) + "]"
+    assert expected_bands_str in evalscript
 
 
 @pytest.mark.parametrize(
     "data_collection, bands, meta_bands",
     [
-        (DataCollection.LANDSAT_TM_L2, None, ["BQA"]),
+        (DataCollection.LANDSAT_TM_L2, ["B05", "B04", "B03", "B02"], ["BQA"]),
         (DataCollection.SENTINEL2_L2A, ["B04", "B03", "B02"], ["CLP", "SCL"]),
     ],
+    ids=str,
 )
-def test_merged_output(data_collection: DataCollection, bands: Optional[List[str]], meta_bands: List[str]) -> None:
+def test_merged_output(data_collection: DataCollection, bands: List[str], meta_bands: List[str]) -> None:
     merged_output = "merged_bands"
     evalscript = generate_evalscript(
         data_collection=data_collection, bands=bands, meta_bands=meta_bands, merged_output=merged_output
     )
-    *_, output_spec = parse_evalscript_info(evalscript)
 
-    num_bands = len(data_collection.bands) if bands is None else len(bands)
-    depth = [depth for name, depth, _ in output_spec if name == merged_output][0]
+    expected_bands_output_spec = "{ " + f'id: "{merged_output}", bands: {len(bands)}'
+    assert expected_bands_output_spec in evalscript
 
-    assert len(output_spec) == 1 + len(meta_bands)
-    assert int(depth) == num_bands
+    expected_bands_return_spec = f"{merged_output}: [" + ", ".join(f"sample.{b}" for b in bands) + "]"
+    assert expected_bands_return_spec in evalscript
 
 
-@pytest.mark.parametrize("merged_output", [None, "merged_bands"])
-@pytest.mark.parametrize("use_dn", [True, False])
-def test_sample_type(merged_output: Optional[str], use_dn: bool) -> None:
+def test_units() -> None:
+    bands = ["B04", "B03", "B02", "B02"]
+    data_collection = DataCollection.SENTINEL2_L1C
+
+    evalscript_dn = generate_evalscript(data_collection=data_collection, bands=bands)
+    assert evalscript_dn.count('"DN"') == len(bands)
+    assert evalscript_dn.count('"REFLECTANCE"') == 0
+
+    evalscript_reflectance = generate_evalscript(data_collection=data_collection, bands=bands, use_dn=False)
+    assert evalscript_reflectance.count('"REFLECTANCE"') == len(bands)
+    assert evalscript_reflectance.count('"DN"') == 0
+
+
+@pytest.mark.parametrize("use_dn", [True, False], ids=lambda x: f"use_dn: {x}")
+def test_sample_type(use_dn: bool) -> None:
+    data_collection = DataCollection.SENTINEL2_L1C
+    evalscript = generate_evalscript(data_collection=data_collection, use_dn=use_dn)
+
+    expected_uint_count = len(data_collection.bands) if use_dn else 0
+    assert evalscript.count('"UINT16"') == expected_uint_count
+
+    expected_float_count = len(data_collection.bands) if not use_dn else 0
+    assert evalscript.count('"FLOAT32"') == expected_float_count
+
+
+@pytest.mark.parametrize("use_dn", [True, False], ids=lambda x: f"use_dn: {x}")
+def test_sample_type_merged(use_dn: bool) -> None:
+    merged_output = "merged_bands"
```

Comment:
what does this do?

Reply:
Generally I had this pattern for cases where I would be using the variable in multiple places, but I see here I just copied the usage. Will just move it inline.

Reply:
Yes. better name suggestions are welcome.

This functionality is sometimes needed when you don't want to download separate bands (in s2cloudless). It could even be useful in eo-grow, where we could remove the extra functionality to explode bands perhaps.

Reply:
so this is both a flag and a name of the output file? gonna need to mull this over :thinking:

Reply:
another option would be to use two parameters:
- `merge_bands_output: bool = False`
- `merged_bands_output_id: str = "bands"`

where the name is made parametrizable, but set to some default, so the user would more or less work only with the boolean flag. Again, accepting better names

Reply:
perhaps a `merged_bands_output` is sufficient for this. It is more specific than `merged_output` so I suspect people will read the docs on what it does.

Reply:
agreed, done.